docs/v7_migration.rst

```diff
@@ -3,11 +3,69 @@
 Migrating your code from v6 to v7
 =================================
 
-web3.py follows `Semantic Versioning <http://semver.org>`_, which means
-that version 7 introduced backwards-incompatible changes. If your
-project depends on web3.py ``v7``, you'll probably need to make some changes.
+web3.py follows `Semantic Versioning <http://semver.org>`_, which means that
+version 7 introduced backwards-incompatible changes. If you're upgrading from
+web3.py ``v6`` or earlier, you can expect to need to make some changes. Refer
+to this guide for a summary of breaking changes when updating from ``v6`` to
+``v7``. If you are more than one major version behind, you should also review
+the migration guides for the versions in between.
+
+
+Provider Updates
+~~~~~~~~~~~~~~~~
+
+
+WebSocketProvider
+`````````````````
+
+``WebsocketProviderV2``, introduced in web3.py ``v6``, has taken priority over the
+legacy ``WebsocketProvider``. The ``LegacyWebSocketProvider`` has been deprecated in
+``v7`` and is slated for removal in the next major version of the library. In summary:
+
+- ``WebsocketProvider`` -> ``LegacyWebSocketProvider`` (and deprecated)
+- ``WebsocketProviderV2`` -> ``WebSocketProvider``
+
+If migrating from ``WebSocketProviderV2`` to ``WebSocketProvider``, you can expect the
+following changes:
+
+- Instantiation no longer requires the ``persistant_websocket`` method:
+
+  .. code-block:: python
+
+      # WebsocketsProviderV2:
+      AsyncWeb3.persistent_websocket(WebsocketProviderV2('...'))
+
+      # WebSocketProvider:
+      AsyncWeb3(WebSocketProvider('...'))
+
+- Handling incoming subscription messages has a more flexible namespace and method name,
+  given that it now supports the AsyncIPCProvider in addition to the WebSocketProvider:
+ 
+  .. code-block:: python
+
+    # WebsocketsProviderV2:
+    async for message in w3.ws.listen_to_websocket():
```

Comment:
This isn't the case in `v2`. It is also `process_subscriptions()`. I think the verbiage above was changed but the code was not?

Reply:
Ah, "namespace and method name" should be changed to "namespace" only in the description above to reflect: `ws` -> `socket`.

Reply:
changed in [v6.15.0](https://web3py.readthedocs.io/en/stable/releases.html#web3-py-v6-15-0-2024-01-25), so if you kept up with latest versions, then you're right